lib/multi_range.rb

```diff
@@ -22,20 +22,16 @@ class MultiRange
 
   attr_reader :ranges
 
-  def initialize(ranges)
+  def initialize(ranges, sort_ranges: true)
     if ranges.is_a? MultiRange
       @ranges = ranges.ranges
-      @is_float = ranges.is_float?
     else
-      @ranges = ranges.map{|s| s.is_a?(Numeric) ? s..s : s }.sort_by(&:begin).freeze
-      @is_float = @ranges.any?{|range| range.begin.is_a?(Float) || range.end.is_a?(Float) }
+      @ranges = ranges.map{ |s| s.is_a?(Numeric) ? s..s : s }
+      @ranges = @ranges.sort_by(&:begin) if sort_ranges
+      @ranges = @ranges.freeze
```

Comment:
MINOR: We don't have to reassign the value here.
```suggestion
      @ranges.freeze
```